src/app/views/homepage/edit/EditHomepageItem.test.js

```diff
@@ -70,3 +140,135 @@ describe("event handlers", () => {
+        expect(wrapper.state("isCreatingImageRecord")).toBe(false);
+
+        // Tests that state is set correctly before asynchronous requests have finished
+        wrapper.instance().createImageRecord();
+        expect(wrapper.state("isCreatingImageRecord")).toBe(true);
+    });
+
+    it("updates state to show it has created image record", async () => {
+        // Tests that state is set correctly after asynchronous requests were successful
+        await wrapper.instance().createImageRecord();
+        expect(wrapper.state("isCreatingImageRecord")).toBe(false);
+        expect(wrapper.state("image")).toBe("test-image-id");
+        expect(wrapper.state("imageState")).toBe("created");
+    });
+
+    it("updates isCreatingImageRecords state correctly on failure to fetch data for all datasets", async () => {
+        image.create.mockImplementationOnce(() => Promise.reject({ status: 500 }));
+        await wrapper.instance().createImageRecord();
+        expect(wrapper.state("isCreatingImageRecord")).toBe(false);
+    });
+
+    it("errors cause notification", async () => {
+        image.create.mockImplementationOnce(() => Promise.reject({ status: 404 }));
+        await wrapper.instance().createImageRecord();
+        expect(mockNotifications.length).toBe(1);
+    });
+});
+
+describe("add upload to image record", () => {
+    const wrapper = mount(<EditHomepageItem {...successRouteProps} />);
+    it("updates isUpdatingImageRecord state to show it's creating image record", () => {
+        expect(wrapper.state("isUpdatingImageRecord")).toBe(false);
+
+        // Tests that state is set correctly before asynchronous requests have finished
+        wrapper.instance().addUploadToImageRecord(mockImage.id, mockImage.upload.path);
+        expect(wrapper.state("isUpdatingImageRecord")).toBe(true);
+    });
+
+    it("updates isUpdatingImageRecord state to show it has created image record", async () => {
+        // Tests that state is set correctly after asynchronous requests were successful
+        await wrapper.instance().addUploadToImageRecord(mockImage.id, mockImage.upload.path);
+        expect(wrapper.state("isUpdatingImageRecord")).toBe(false);
+    });
+
+    it("updates isCreatingImageRecords state correctly on failure to fetch data for all datasets", async () => {
+        image.update.mockImplementationOnce(() => Promise.reject({ status: 500 }));
+        await wrapper.instance().addUploadToImageRecord(mockImage.id, mockImage.upload.path);
+        expect(wrapper.state("isUpdatingImageRecord")).toBe(false);
+    });
+
+    it("errors cause notification", async () => {
+        image.update.mockImplementationOnce(() => Promise.reject({ status: 404 }));
+        await wrapper.instance().addUploadToImageRecord(mockImage.id, mockImage.upload.path);
+        expect(mockNotifications.length).toBe(1);
+    });
+});
+
+describe("get image upload ", () => {
```

Comment:
If we did change up the way we're storing the current image/import status, we'd need to update these tests too. `expect(wrapper.state("currentImageImportStatus").toEqual("Image is loading")` etc